### ISSUE TITLE
Initial check list tasks to do

### ISSUE THREAD

**Issue by evandrocoan** [evandrocoan/ObjectBeautifier#1]
### Initial check list tasks to do

1. [ ] Make sure to create Unit Tests for each scope definition you create so this way, the scope definitions are like the formatting settings, for each one of them, there are several unit tests!

**Comment by evandrocoan** [comment]
### Initial UML Diagram and Design:

![1](https://cloud.githubusercontent.com/assets/5332158/21577488/a4ffcf0e-cf43-11e6-9f69-7ecf7e4694f8.png)

**Comment by evandrocoan** [comment]
### The formatting is performed abstracted of language.

The languages still existing, however the formatting will be based on languages but in visual formatting concepts. The formatting based in visual formatting concepts. There will not be a setting to add spaces in paren for java, c++, etc. There will be a setting to add spaces in paren for specific regex and/or inclusion and exclusion laws formed on that particular setting file, together within its Unit Tests.

Basically a setting file it will be very big and complex and only will treat a specific kind of visual formatting concept. Therefore to simply allow how to choose and setup a visual style for your own taste or needs, not all settings will be loaded to be processed and applied to parse the source code. Moreover not loading any settings file will imply on a zero-configuration, i.e., nothing will be changed on the source code after the parsing.

**Comment by evandrocoan** [comment]
### The General Scoping Rules

Instead of always to writing the rule for opening and defining a commentary scope rule on every settings files, create it at only one file per language characteristic scoping rules as aliases which can be used across all settings files. We will not create one scoping setting file per language because it would be pointless as the several languages uses the exactly construct tool. For example, the C++ block comments constructs `/* */` are the same on C, Java and Pawn. So it would be created one file per scoping concept as `block comments`. On this file all the scoping rules for all languages will be defined and used across the configurations files.
```pawn
// This is the name used to reference this scope around the settings files.
Scope Name:
c++_block_comment

// This
Scope Expression:
```
